### PR TITLE
Cleanup database proposal creation code

### PR DESCRIPTION
**Pull request by jsuchome**

### CODE REVIEW

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -2231,39 +2231,39 @@ function custom_configuration
...
+            if [[ $want_database_sql_engine ]] ; then
+                proposal_set_value database default "['attributes']['database']['sql_engine']" "'$want_database_sql_engine'"
+            fi
+            # For SOC7 we introduced transitional role called mysql-server that's gonna be used during the upgrade
+            # Users deploying SOC7 with MariaDB must use this one and not database-server
+            # However to not break GM deployments, we should check if the role is actually present.
+            if iscloudver 7 && [ -e "/opt/dell/chef/data_bags/crowbar/migrate/database/109_separate_db_roles.rb" ]; then
+                # In this branch, if want_database_sql_engine is not set, mysql is the default
+                if [[ $hacloud = 1 ]] ; then
+                    if [[ "$want_database_sql_engine" != "postgresql" ]] ; then
```
Comment (rhafer) [inline scripts/qa_crowbarsetup.sh]: If mysql is going to be deployed here, why do we need to set any drdb related attributes?

Also as we are past the 109 migration those attributes would live in `database.postgresql.ha....` wouldn't they?

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -2231,39 +2231,39 @@ function custom_configuration
...
                         proposal_set_value database default "['attributes']['database']['ha']['storage']['mode']" "'drbd'"
                         proposal_set_value database default "['attributes']['database']['ha']['storage']['drbd']['size']" "$drbd_database_size"
+                        proposal_set_value database default "['deployment']['database']['elements']['mysql-server']" "['cluster:$clusternamedata']"
+                    else
```
Comment (rhafer) [inline scripts/qa_crowbarsetup.sh]: I'd probably add a comment, that this branch means we're going to deploy postgresql

File: scripts/qa_crowbarsetup.sh (hunk 1 above)
Comment (jsuchome) [reply scripts/qa_crowbarsetup.sh]: True. I wonder if this case was incorrectly there before...

File: scripts/mkcloud [hunk 3]
```diff
@@ -109,7 +109,7 @@ iscloudver 7plus && : ${controller_node_memory:=12582912}
 : ${lonelynode_hdd_size:=20}
 : ${ironicnode_hdd_size:=20}
 
-if [[ $hacloud ]] && [[ "$want_database_sql_engine" != "mysql" ]]; then
+if [[ $hacloud ]] && [[ "$want_database_sql_engine" == "postgresql" ]]; then
```
Comment (rhafer) [inline scripts/mkcloud]: As you noted already this doesn't work on Cloud 7 GA, when  "$want_database_sql_engine" is not set. I'd be fine with fixing this in a followup.

But I think we also still care about Cloud6, don't we?

**Pull request merged**